Sources/Plasma/PubUtilLib/plUnifiedTime/plUnifiedTime.cpp

```diff
@@ -107,10 +112,22 @@ void plUnifiedTime::ToCurrentTime()
     struct timespec ts;
 
 #if defined(HS_BUILD_FOR_APPLE)
-    // timespec_get is only supported since macOS 10.15,
-    // but clock_gettime exists since macOS 10.12.
-    int res = clock_gettime(CLOCK_REALTIME, &ts);
-    hsAssert(res == 0, "clock_gettime failed");
+#   ifdef HAVE_BUILTIN_AVAILABLE
+    if (__builtin_available(macOS 10.12, *)) {
```

Comment:
Since we don't actually need nanosecond precision, you can just use `gettimeofday` unconditionally and skip the availability check. https://github.com/H-uru/Plasma/pull/1396#discussion_r1208455783

Reply:
There are a few sources (including code-scanning tools) asserting that `gettimeofday` is semi-deprecated and should not be used in new code (but is part of the POSIX standard so it won't actually get removed). Since we already have the `clock_gettime` implementation and it's easy to guard, it seems worth keeping for newer systems.

Reply:
Ugh... In that case, could we at least remove the `clock_gettime` branch? If we have to add an availability check anyway, we can use `timespec_get` where available, and fall back to `gettimeofday` for less recent macOSes/SDKs.

Reply:
Sure. Remind me, `timespec_get` is 10.15 and newer?

Reply:
Yep:

https://github.com/H-uru/Plasma/blob/1c4209cb2e5b5852bfec11d0cab7a5be6eaf72d2/Sources/Plasma/PubUtilLib/plUnifiedTime/plUnifiedTime.cpp#L110-L111